Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MetsXmlElementAccess.java

```diff
@@ -128,76 +102,49 @@ public Workpiece() {
+                            UseXmlAttributeAccess::getMediaVariant))
                 : new HashMap<>();
-        Optional<StructMapType> optionalPhysical = getStructMapsStreamByType(mets, "PHYSICAL").findFirst();
-        Map<String, Set<MediaUnit>> mediaUnitsMap;
-        if (!optionalPhysical.isPresent()) {
-            mediaUnitsMap = Collections.emptyMap();
-        } else {
-            List<DivType> physicalDivs = optionalPhysical.get().getDiv().getDiv();
-            Map<String, MediaUnit> divIDsToMediaUnits = new HashMap<>((int) Math.ceil(physicalDivs.size() / 0.75));
-            for (DivType div : physicalDivs) {
-                MediaUnit mediaUnit = new MediaUnit(div, mets, mediaVariants);
-                mediaUnits.add(mediaUnit);
-                divIDsToMediaUnits.put(div.getID(), mediaUnit);
-            }
-            mediaUnitsMap = mets.getStructLink().getSmLinkOrSmLinkGrp().parallelStream()
-                    .filter(SmLink.class::isInstance).map(SmLink.class::cast).collect(
-                        new MultiMapCollector<>(SmLink::getFrom, smLink -> divIDsToMediaUnits.get(smLink.getTo())));
+        Optional<StructMapType> optionalPhysicalStructMap = getStructMapsStreamByType(mets, "PHYSICAL").findFirst();
+        List<DivType> physicalDivs = optionalPhysicalStructMap.isPresent()
+                ? optionalPhysicalStructMap.get().getDiv().getDiv()
+                : Collections.emptyList();
+        Map<String, FileXmlElementAccess> divIDsToMediaUnits = new HashMap<>((int) Math.ceil(physicalDivs.size() / 0.75));
+        for (DivType div : physicalDivs) {
+            FileXmlElementAccess fileXmlElementAccess = new FileXmlElementAccess(div, mets, useXmlAttributeAccess);
+            workpiece.getMediaUnits().add(fileXmlElementAccess.getMediaUnit());
+            divIDsToMediaUnits.put(div.getID(), fileXmlElementAccess);
         }
-        structure = getStructMapsStreamByType(mets, "LOGICAL")
-                .map(structMap -> new Structure(structMap.getDiv(), mets, mediaUnitsMap)).collect(Collectors.toList())
-                .iterator().next();
-    }
-
-    /**
-     * Returns the media units of this workpiece.
-     * 
-     * @return the media units
-     */
-    @Override
-    public List<FileXmlElementAccessInterface> getFileGrp() {
-        return mediaUnits;
-    }
-
-    /**
-     * Returns the edit history. The head of each METS file has space for
-     * processing notes of the individual editors. In this way, the processing
-     * process of the digital representation can be understood.
-     * 
-     * @return the edit history
-     */
-    @Override
-    public List<AgentXmlElementAccessInterface> getMetsHdr() {
-        return editHistory;
+        if (mets.getStructLink() == null) {
+            mets.setStructLink(new StructLink());
+        }
+        Map<String, Set<FileXmlElementAccess>> mediaUnitsMap = mets.getStructLink().getSmLinkOrSmLinkGrp().parallelStream()
```

Comment:
sorry again, this stream is too long to understand

Reply:
Same as above, this is [already existing](https://github.com/kitodo/kitodo-production/blob/master/Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/Workpiece.java#L162-L164) code and unrelated with the current change. If you want to, we can change it, but I would like to do it in a separate pull request.